src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/deciders/CacheHealthDecider.java

```diff
@@ -59,10 +69,16 @@ public CacheHealthDecider(
     this.fieldDataCacheClusterRca = fieldDataCacheClusterRca;
     this.shardRequestCacheClusterRca = shardRequestCacheClusterRca;
     this.cacheTypeBaseClusterRcaMap =
-        ImmutableMap.<ResourceEnum, BaseClusterRca>builder()
-            .put(ResourceEnum.SHARD_REQUEST_CACHE, shardRequestCacheClusterRca)
-            .put(ResourceEnum.FIELD_DATA_CACHE, fieldDataCacheClusterRca)
-            .build();
+            ImmutableMap.<ResourceEnum, BaseClusterRca>builder()
+                    .put(ResourceEnum.SHARD_REQUEST_CACHE, shardRequestCacheClusterRca)
+                    .put(ResourceEnum.FIELD_DATA_CACHE, fieldDataCacheClusterRca)
+                    .build();
+    this.heapUsageBucketCalculator = new BasicBucketCalculator(HEAP_USAGE_MAP);
+    try {
+      this.heapUsageBucketCalculator = rcaConf.getBucketizationSettings(HEAP_TUNABLE_NAME);
```

Comment:
I am thinking that this bucket calculator should be a part of the rca and not the decider (also see comment about moving HeapUsage to an rca instead of NodeCofigCollector below).

This is because these bucketization values are closely linked to the rca or metric, rather than the decider. Suppose there are 3 different metrics that are being bucketized, say young-gen-util, old-gen-util and cpu-util. If decider applies bucketization, then each decider must remember the `tunableName` for each bucketization rcaconf setting. Whereas, if rcas apply it, they need to just remember their own `tunableName` values.

We should just pass rcaConf to the rcas and let them return the usage buckets. They can also chose the bucket calculator relevant to them, and implement some interface (`Bucketizable` ?) which exposes a `getUsageBucket` method.